backend/app/api/endpoints/admin/device_monitor.py

```
@@ -708,15 +709,27 @@ async def restart_device(
+    try:
+        result = await restart_admin_device(db, user_id, device_id)
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.exception(
+            f"[Admin Device Restart] Error: device_id={device_id}, error={e}"
+        )
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"Failed to trigger restart: {str(e)}",
+        )
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)